### PR TITLE
Implement configprovider interface

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR provides the skeleton for implementing the configprovider interface. This custom implementation of configprovider will then be used to enhance the config provided.

### CODE REVIEW

File: comp/otelcol/otlp/components/pipeline/provider/go.mod [hunk 1]
```diff
@@ -0,0 +1,98 @@
+module provider
+
+go 1.21.6
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/pipeline/provider/go.mod:3]: use 1.21.0

File: comp/otelcol/otlp/components/pipeline/provider/provider.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	// TODO: modify conf (add datadogconnector if not present ...etc)
+	//
+	return conf, nil
+}
+
+// Watch is a no-op which returns a nil chan.
+func (cp *configProvider) Watch() <-chan error {
+	return nil
+}
+
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/pipeline/provider/provider.go:65]: may be return empty channel

**Review by dineshg13 — APPROVED** [review]

File: comp/otelcol/otlp/components/pipeline/provider/provider.go (hunk 2 above)
Comment (mackjmr) [reply comp/otelcol/otlp/components/pipeline/provider/provider.go:65]: read or write on a nil channel will block indefinitely, so I think this is OK for our use case here ?

File: comp/otelcol/otlp/components/pipeline/provider/provider.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package provider
```
Comment (songy23) [inline comp/otelcol/otlp/components/pipeline/provider/provider.go:1]: Missing license header

File: comp/otelcol/otlp/components/pipeline/provider/provider_test.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
+package provider
```
Comment (songy23) [inline comp/otelcol/otlp/components/pipeline/provider/provider_test.go:1]: ditto

**Pull request merged**